### PR TITLE
Build: List jQuery Core module files if present

### PR DESCRIPTION
**Pull request by mgol**

### CODE REVIEW

**Comment by mgol** [comment]
Note that this does not add factory files - they are meant for Node.js so we don't plan to upload them to the CDN, just make them available in the npm package (and the bower one but that's just because contents are the same).

**Comment by mgol** [comment]
For now, the module files only exist for the Git builds as they will only be available starting with jQuery 4.0.0. Those already exist so you can check them.

**Comment by Krinkle** [comment]
For "slim", we label it "slim" and "slim minified", with a space instead of parenthesis. I tend to avoid parenthesis in navigational links when possible, to make it feel less complex to the reader.

Today:

> [uncompressed](#), [minified](#), [slim](#), [slim minified](#).

If we follow the wording for "slim":

> [uncompressed](#), [minified](#), [slim](#), [slim minified](#), [module](#), [module minified](#), [slim module](#), [slim module minifed](#).

With more labels and parenthesis:

> [uncompressed](#), [minified](#), [slim (uncompressed)](#), [slim (minified)](#), [module (uncompressed)](#), [module (minified)](#), [slim module (uncompressed)](#), [slim module (minifed)](#).

Vertical separation:

> script: [uncompressed](#), [minified](#), [slim](#), [slim minified](#)
> module: [uncompressed](#), [minified](#), [slim](#), [slim minified](#).


> Note that this does not add factory files - they are meant for Node.js so we don't plan to upload them to the CDN, just make them available in the npm package […]

Hm.. in that case, do we need the above logic? Or would we only need the `-git` links in that case, which are a separate part of the patch afaik. The above is only applied to CDN and stable releases, right?

**Comment by mgol** [comment]
@Krinkle We don't plan to upload _factory_ files:
* https://github.com/jquery/jquery/blob/ace646f6e83e653f666ba715c200739f1cbdba52/package.json#L30-L37

or Node module wrapper files:
* https://github.com/jquery/jquery/blob/ace646f6e83e653f666ba715c200739f1cbdba52/package.json#L11
* https://github.com/jquery/jquery/blob/ace646f6e83e653f666ba715c200739f1cbdba52/package.json#L22

We do plan to upload module files:
* https://github.com/jquery/jquery/blob/ace646f6e83e653f666ba715c200739f1cbdba52/package.json#L14-L15
* https://github.com/jquery/jquery/blob/ace646f6e83e653f666ba715c200739f1cbdba52/package.json#L25-L26

so yes, this additional logic is needed.

The list gets quite long now so I like the vertical separation idea. I'll need to refactor the code a bit for that.

**Review by timmywil — APPROVED** [review]

File: templates/jquery-git.hbs [hunk 1]
```diff
@@ -8,6 +8,10 @@
 		<a href="/git/jquery-git.min.js">minified</a>,
 		<a href="/git/jquery-git.slim.js">slim</a>,
 		<a href="/git/jquery-git.slim.min.js">slim minified</a>
```
Comment (Krinkle) [inline templates/jquery-git.hbs:10]: You might want to place a comma after this line for consistency.

Comment (mgol) [reply templates/jquery-git.hbs:12]: This part has changed significantly so the comment is no longer relevant.

**Comment by mgol** [comment]
PR updated. The latest state of the home page limited to the jQuery Core section:
Screenshot 2024-01-27 at 00 45 43

and the jQuery Core page itself:
Screenshot 2024-01-27 at 00 45 27

I changed hyphens to colons universally as hyphens look bad in the multiline setup and IMO it's better to settle on one for consistency.

**Review by timmywil — APPROVED** [review]
LGTM

**Comment by mgol** [comment]
This change is still not deployed at https://releases.jquery.com/jquery/. I tested it locally and it worked. Any ideas?

**Comment by mgol** [comment]
Actually, no commit since https://github.com/jquery/codeorigin.jquery.com/commit/53da21880f4d29f46feccc5577fb36b988b180ab is getting deployed. I posted on `#jquery-infra` on Slack.

**Pull request merged**